internal/controller/resource_manager_test.go

```diff
@@ -239,3 +239,56 @@ func TestProcessMutation_ConstructsMetadataCorrectly(t *testing.T) {
 		t.Fatalf("unexpected metadata in DeleteResource: %#v", got)
 	}
 }
+
+func TestIsExpiredResourceVersionError(t *testing.T) {
+	tests := []struct {
+		name     string
+		err      error
+		expected bool
+	}{
+		{
+			name:     "nil error",
+			err:      nil,
+			expected: false,
+		},
+		{
+			name:     "expired resource version error with exact message from customer",
+			err:      errors.New("code: 410, reason: Expired, message: too old resource version: 1762495487792497000 (1762495505417599006)"),
```

Comment:
You need to build errors with the same types as apimachinery to properly test this.
```suggestion
			err:      errors.NewResourceExpired("too old resource version: 1762495487792497000 (1762495505417599006)"),
```